### PR TITLE
py-optax: add new package

### PR DESCRIPTION
**Pull request by ChristopherChristofi**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-optax/package.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    homepage = "https://github.com/deepmind/optax"
+    pypi = "optax/optax-0.1.7.tar.gz"
+
+    license("Apache-2.0")
+
+    version("0.1.7", sha256="6a5a848bc5e55e619b187c749fdddc4a5443ea14be85cc769f995779865c110d")
+
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-flit-core@3.2:4", type="build")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-optax/package.py]:
```suggestion
    depends_on("py-flit-core@3.2:3", type="build")
```

File: var/spack/repos/builtin/packages/py-optax/package.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class PyOptax(PythonPackage):
+    """A gradient processing and optimisation library in JAX."""
+
+    homepage = "https://github.com/deepmind/optax"
+    pypi = "optax/optax-0.1.7.tar.gz"
+
+    license("Apache-2.0")
+
+    version("0.1.7", sha256="6a5a848bc5e55e619b187c749fdddc4a5443ea14be85cc769f995779865c110d")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-optax/package.py:17]: There is a newer version (0.2.11) available on pypi as well.

Comment (ChristopherChristofi) [reply var/spack/repos/builtin/packages/py-optax/package.py:17]: Hello @manuelakuhn, your mentioned version refers to a different package, opax not optax

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-optax/package.py:17]: ah sorry, then don't mind :-)

**Review by manuelakuhn — APPROVED** [review]
LGTM, unfortunately I can not build it because bazel does not build for me (see #41346)

**Review by adamjstewart — APPROVED** [review]

**Comment by ChristopherChristofi** [comment]
@spackbot run pipeline

**Comment by ChristopherChristofi** [comment]
@spackbot re-run pipeline

**Comment by ChristopherChristofi** [comment]
I am not too sure how to fix this pipeline failure, it appears to have failed due to timeout on re-runs
FYI @adamjstewart, @manuelakuhn

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Pull request merged**